pyiron_workflow/nodes/composite.py

```diff
@@ -26,6 +27,130 @@
+    node: Node, with_values: bool = True, with_default: bool = True
+) -> dict:
+    data = {"inputs": {}, "outputs": {}}
+    is_composite = isinstance(node, Composite)
+    for io_ in ["inputs", "outputs"]:
+        for inp in getattr(node, io_):
+            if is_composite:
+                data[io_][inp.scoped_label] = _extract_data(
+                    inp, with_values=with_values, with_default=with_default
+                )
+            else:
+                data[io_][inp.label] = _extract_data(
+                    inp, with_values=with_values, with_default=with_default
+                )
+    return data
+
+
+def export_node_to_dict(
+    node: Node, with_values: bool = True, with_default: bool = True
+) -> dict:
+    """
+    Export a node to a dictionary.
+
+    Args:
+        node (Node): The node to export.
+        with_values (bool): Whether to include the values of the channels in the
+            dictionary. (Default is True.)
+
+    Returns:
+        dict: The exported node as a dictionary.
+    """
+    data = {"inputs": {}, "outputs": {}, "function": node.node_function}
+    data.update(_io_to_dict(node, with_values=with_values, with_default=with_default))
+    return data
+
+
+def export_composite_to_dict(
+    workflow: Composite, with_values: bool = True, with_default: bool = True
+) -> dict:
+    """
+    Export a composite to a dictionary.
+
+    Args:
+        workflow (Composite): The composite to export.
+        with_values (bool): Whether to include the values of the channels in the
+            dictionary. (Default is True.)
+
+    Returns:
+        dict: The exported composite as a dictionary.
+    """
+    data = {"inputs": {}, "outputs": {}, "nodes": {}, "edges": []}
+    for inp in workflow.inputs:
+        if inp.value_receiver is not None:
+            data["edges"].append(
+                (
+                    f"inputs.{inp.scoped_label}",
+                    _get_scoped_label(inp.value_receiver, "inputs"),
+                )
+            )
+    for node in workflow:
+        label = node.label
+        if isinstance(node, Composite):
+            data["nodes"][label] = export_composite_to_dict(
+                node, with_values=with_values
+            )
+        else:
+            data["nodes"][label] = export_node_to_dict(node, with_values=with_values)
+        for inp in node.inputs:
+            if _is_internal_connection(inp, workflow, "outputs"):
+                data["edges"].append(
+                    (
+                        _get_scoped_label(inp.connections[0], "outputs"),
+                        _get_scoped_label(inp, "inputs"),
+                    )
+                )
+        for out in node.outputs:
+            if out.value_receiver is not None:
+                data["edges"].append(
+                    (
+                        _get_scoped_label(out, "outputs"),
+                        f"outputs.{out.value_receiver.scoped_label}",
+                    )
+                )
```

Comment:
This is super similar to the same scraping for the input value receiver pairs, just a sort of conjugate. I think it should be possible to nicely pull out a `_get_value_synchronization_edges(sender: Node, receiver: Node, io: str)` function